user-profiles.md

```diff
@@ -0,0 +1,49 @@
+# User Profiles
+
+The purpose of this document is to aid in the development of Tekton features by evaluating Tekton features against those who will be using them.
+
+When considering requirements and implementation solutions it's useful look at who uses Tekton to have a better understanding of how something should work. The Tekton user profiles describe the different types of users and contributors to Tekton along with the order of priority they have relative to each other. The ordering is because nothing can share the exact same priority and we want to convey that.
+
+## Preface
+
+* What's described here are user profiles as opposed to [personas](https://en.wikipedia.org/wiki/Persona#In_user_experience_design). Personas are example actors rather than general categories.
+* Kubernetes Cluster Operators are out of scope for this document. A cluster operator is one who manages the operation of a Kubernetes cluster that applications can run in.
+* Application Operators are in scope for this document at a lower priority than Pipeline Operators. An application operator is one who manages the operation of the application in a Kubernetes cluster.
+
+## Profiles
+
+Profiles describe a type of role a user may perform. A real person may perform more than one role and have more than one profile apply to them. How this mapping works between profiles and real people can vary between companies and other organizations. To handle this variation we focus on the user profiles rather than how they may map to people in these different organizations.
+
+### 1. Pipeline Operator
+
+Pipeline operators define, schedule, monitor, and otherwise operate pipelines across one or more Kubernetes clusters. For example, the operation of a set of Kaniko build pipelines and Argo CD pipelines. This is not to be confused with the role of a Kubernetes cluster operator. This is also not to be confused with the application's operator role, who would oversee a product in production.
+
+### 2. Pipeline Distributor
+
+Distributors are people who package an application and application pipelines for someone else to operate. Examples of this would be those who maintain [Catalog Pipelines](https://github.com/tektoncd/catalog) such as for Kaniko and ArgoCD.
+
+### 3. Application Operator and Application Distributor
+
+Application operators take an application and operate it within a Kubernetes cluster. For example, the operation of WordPress and MySQL. This is not to be confused with the role of a Kubernetes cluster operator (see earlier in preface for more detail). Application distributors are people who package an application for someone else to operate. Examples of this would be those who maintain community Helm Charts for WordPress and Datadog. These are the same profiles as seen in Helm's profile document https://github.com/helm/community/blob/master/user-profiles.md
```

Comment:
How would this translate to `Tekton` ? Isn't this role unrelevant to Tekton, or a subset of `Pipeline Distributor` ?

Reply:
This is why I need an external perspective. From Tekton's perspective, are there application developers? I argue they still exist, though Application Developers at most places I have worked previously are at-best apathetic towards CI/CD currently, aside from making their own new workflows (eg. for a new product team), and making sure the build goes through.

Let me give the example of Jenkins X as an Application Developer-centric approach to building an experience on Tekton. This is a big consideration to me. Will Tekton care about allowing Application Developers a good experience with Tekton? Jenkins/Hudson/CircleCI/TravisCI always at least gave webhooks and other types of buttons for kicking off build pipelines, as example. Does Tekton or its CLI make instantiating a build system simple enough an App Dev can hook it into Visual Studio Code?

Specifically, I like the concept of expanding this direction, though it could be overlapped with external projects, or split into a separate set of projects that Tekton never thinks about directly

Reply:
I do think it is at least a separate role. One distinction of the role is "user-of, not developer-of anything to do with Tekton"? Might rephrase the role description or give it a different name that encompasses "mostly-blissfully-ignorant of internal details of Tekton, yet still wants to use it, and while technically capable of learning and passionate about learning, is time-constrained against learning the inner workings of K8S/Tekton - for now". 

Like, lets get 'em hooked before asking AppDevs to read a dictionary or two of terms we invented for Tekton and ask them to understand how Knative's injection infrastructure works. I think there is a valued user to be caught.

:thinking: Though, if that type of user is out of scope for Tekton, let's at least call it out

Reply:
If an Application Operator uses a system other than TektonCD for shipping updates, what do we care about their opinion of the TektonCD system. Is this more like the question?

If someone packages an Application using myriad systems, though never uses Tekton, then they are perhaps irrelevant, as you mention.

I might call these out as out of scope profiles, excepting those that use Tekton for completing these roles. If this does not look right then they can also be removed.

Moved these to the Not In Scope section

Reply:
Though, there is a question: Do we want to support Application Operators (to convince them to start using Tekton)? It seems a bit off to me to simply drop Operator support, at first glance. I think the first example of an Operator task might be "Configure Logging" for some deployment or "monitor CPU". Is Tekton neutral on configuring logging for the final deployed environment? Obviously, Tekton wants to export logs from the intermediate steps or from running the pipelines themselves, though does Tekton care or need to care about what happens in production? Is that handed off to other tools? Does Tekton consider itself managing release environments and deployed artifacts? In a K8S world, Tekton is absolutely generating the images, then deploying those images. 

In general, I have considered the principle of "If I create it, I destroy it" or at least offer some tracking mechanisms for others to clean it up later.

Leaving in the Not In Scope section where it can get discussion